### DISCUSSION TITLE
#5 📑 Weekly Report 5

### DISCUSSION THREAD

**Discussion by ins305** [SE-TINF22B6/Statify#48] [Blog]
Welcome back!👋
In the 5th week we completed some tasks which help us defining the structure of our project. We talked about classes we will be using in our application and drew some more diagrams. 

## Classes
Last week in our CRC workshop we discussed which classes our application needs. With the results we created a [UML class diagram](https://github.com/SE-TINF22B6/Statify/blob/main/Documentation/diagrams/UML_Class_Diagram.svg).

![UML_Class_Diagram](https://github.com/SE-TINF22B6/Statify/assets/64202248/ba9ed673-d3c4-4bd0-87dc-bcabbbb2e1a0)

As we are going to use data from the Spotify API to generate statistics about your listening habits, we have several classes (Track, Artist, Playlist, AudioFeatures) which will be populated with the data we get from Spotify. Also, we included classes for different types of statistics, as well as an interface they all implement. Additionally, there are some classes which will be used in the backend for example to access the database and send requests to the Spotify API. Also important is the "User" class which handles the users data.

## ER diagram
Another task we completed was to create an [ER diagram](https://github.com/SE-TINF22B6/Statify/blob/main/Documentation/diagrams/ER_Diagram.svg). Here you can see which data will be stored in our relational database.

![ER_Diagram](https://github.com/SE-TINF22B6/Statify/assets/57537992/1520fe2d-00a6-431a-b83a-ced6e2cfc95d)

## Sequence Diagrams
[Sequence diagram architecture](https://github.com/SE-TINF22B6/Statify/blob/main/Documentation/diagrams/Sequence_diagram_architecture.pdf)
[Sequence diagram technical](https://github.com/SE-TINF22B6/Statify/blob/main/Documentation/diagrams/Sequence_diagram_technical.png)

**Comment by Umbranecis** [comment]
I really like the detailed Overview that makes your Project clearly understandable.
I also really like how you shor a clear structure in your classes and their interaction. Looks very clean

However, I think you could add a little more info on where to start when reading your diagram, that would make it easier.

Also, as a sidenote, in Github it's difficult to see the diagram because the black arrows are barely visible on black background. I had to export the diagram to view it. Maybe next time insert it with a white background.

**Comment by derrrobin** [comment]
Hey, 
I appreciate the thorough project overview, which greatly enhances the clarity of your work. The organized structure of your classes and their interactions is commendable and contributes to a clean presentation.

Nevertheless, I also suggest providing additional guidance on where to commence when interpreting your diagram, as this would enhance accessibility for readers.

On a side note, accessing the diagram on GitHub proved challenging due to the black arrows against a dark background. I had to export the diagram to view it properly. For future reference, consider using a white background to improve visibility directly on the platform.

Your CookPit Team
